### PR TITLE
Cleanup includes

### PR DESCRIPTION
**Pull request by tobim**
This is done by manually integrating suggestions from iwyu. This change will also simplify running `iwyu` and `clang-tidy` via CMake by disabling these tools for the CAF and broker submodules.

This change also moves the CAF and broker related CMake code into `aux/CMakeLists.txt`.

### CODE REVIEW

File: libvast/src/table_slice.cpp [hunk 1]
```diff
@@ -255,7 +255,7 @@ std::pair<table_slice_ptr, table_slice_ptr> split(const table_slice_ptr& slice,
   // Create second table slice.
   select(xs, slice, make_ids({{mid, last}}));
   VAST_ASSERT(xs.size() == 2);
-  return {std::move(xs.front()), std::move(xs.back())};
+  return {xs.front(), xs.back()};
```
Comment (dominiklohmann) [inline libvast/src/table_slice.cpp]: Changed by accident?

File: vast.imp [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[
+  { ref: "caf.imp" }
+]
```
Comment (dominiklohmann) [inline vast.imp]: Added to git by accident?

Comment (tobim) [reply vast.imp]: This is an _iwyu-mapping-file_, together with the referenced `caf.imp` they contain out-of-band directives for telling the tool which header is to be used as the public source of a definition.

**Comment by lava** [comment]
> This change will also simplify running iwyu and clang-tidy via CMake by disabling these tools for the CAF and broker submodules.

How is this done?

**Comment by tobim** [comment]
> > This change will also simplify running iwyu and clang-tidy via CMake by disabling these tools for the CAF and broker submodules.
> 
> How is this done?

The trick is to set the target properties `CXX_CLANG_TIDY` and `CXX_INCLUDE_WHAT_YOU_USE` to the empty list each. See https://cmake.org/cmake/help/latest/prop_tgt/LANG_INCLUDE_WHAT_YOU_USE.html for details.

**Comment by tobim** [comment]
I'll close this for now, parts of it should be salvageable in more focused PRs later. I'd like to follow through with the changes to `fwd.hpp` at least.

**Pull request closed without merging**